src/lib/io/sol-gpio-common.c

```diff
@@ -74,3 +77,39 @@ sol_gpio_open(uint32_t pin, const struct sol_gpio_config *config)
 
     return gpio;
 }
+
+SOL_API enum sol_gpio_edge
+sol_gpio_edge_from_str(const char *edge)
+{
```

Comment:
If you're referring to https://github.com/solettaproject/soletta/pull/1767#discussion_r57982171, then that's not the same as what I'm saying. What I'm saying is that you don't need the functions ```sol_gpio_edge_from_str()``` and ```sol_gpio_edge_to_str()``` at all. You can write the lookup in both directions JS with a lot less code. See my comment [here](https://github.com/solettaproject/soletta/pull/1796/files#r58203275).